### PR TITLE
Jetpack Connect: Move input redirections out of redux

### PR DESCRIPTION
**Pull request by sirreal**
The PR moves redirection actions which are only relevant for the site input step out of redux and into the component.

It also colocates some constants in a new constants file.

## Testing
1. Start with a new site with no Jetpack installed
1. Ensure installation works as before
1. Ensure activation works as before
1. Ensure auth works as before (moves to auth step correctly).
1. Ensure `goToPlans` works
   * enter a url for a site you already own on a free plan
   * You'll be redirected to `/jetpack/connect/plans` immediately

### CODE REVIEW

File: client/jetpack-connect/main.jsx [hunk 1]
```diff
@@ -95,12 +97,12 @@ class JetpackConnectMain extends Component {
 		if (
 			this.getStatus() === 'notConnectedJetpack' &&
 			this.isCurrentUrlFetched() &&
-			! this.props.jetpackConnectSite.isRedirecting
+			! this.redirecting
```
Comment (sirreal) [inline client/jetpack-connect/main.jsx]: Tempting to remove, but could change behavior based on early returns.

**Comment by sirreal** [comment]
Adding some unit tests…

File: client/jetpack-connect/main.jsx [hunk 2]
```diff
@@ -112,6 +114,50 @@ class JetpackConnectMain extends Component {
 
 	dismissUrl = () => this.props.dismissUrl( this.state.currentUrl );
 
+	makeSafeRedirectionFunction( func ) {
```
Comment (seear) [inline client/jetpack-connect/main.jsx]: Nice.

**Comment by sirreal** [comment]
Thanks @seear!

>  Going to tackle goBackToWpAdmin() next..?

It's certainly on my list 😁

**Pull request merged**